feat(autocomplete): pass md-input-max/min-length to the input with no…

theLufenk opened the pull request:
…n-floating label

Adds the md-input-minlength and md-input-maxlength attributes as ng-minlength and ng-maxlength to the input element of the autocomplete directive when a floating label is not used.
A use case : It would help in validating the input when the value of Search Text is being used if no matches are found.

theLufenk commented:
I believed it would be taken up with #4492 . However it wasn't.
As of now, there is no way to add length validation to the text input element.

The value of Search Text is exposed , hence it can be used in a scenario when we want the autocomplete to serve as both a suggestion list as well as text input .

DevVersion commented:
Thanks for the pull request - The reason why the `ng-min/max-length` attributes are not supported a normal autocomplete is that we don't support `ng-messages` for a normal autocomplete.

My question here is, how will you validate the "underlaying" input?

theLufenk commented:
The property is exposed as [FORM_NAME].[MD_INPUT_NAME].$error.maxlength nonetheless.

DevVersion commented:
@theLufenk Ah I see. From this perspective it makes sense to support those `validator` attributes for the normal autocomplete.

Can you please some tests for those two attributes?

theLufenk commented:
Added tests. The spec was missing tests for md-input-minlength. Added them too.
I suppose checking for presence of ng-maxlength/ng-minlength attribute in the render input element and $valid state of container forms would be sufficient ?

DevVersion commented:
@theLufenk Perfect! I think the tests which are only checking for the `attribute` on the input are not necessary when having the tests for the `$valid` state.

> If you have time it would be good to have them removed (those are unnecessary and bloating the specs)

theLufenk commented:
Done and rebased.
